app/controllers/account/tasks_controller.rb

```diff
@@ -134,6 +132,27 @@ def report
       disposition: 'attachment'
   end
 
+  def new_task_from_calendar
+    @task = Task.new
+    @task.due_date = params[:due_date]
+
+    respond_to :js
+  end
+
+  def create_task_from_calendar
+    @task = Task.new(calendar_task_params)
+    if @project = current_user.available_projects.where(id: calendar_task_params[:project_id]).first
```

Comment:
don't do two operations in one line 

first find project
second check if need

it's much better for readability